### PR TITLE
Replace max-age with Etag checksum check to fix attachment preview caching

### PR DESCRIPTION
**Pull request by SammyIsConfused**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] documentation is changed or added (in-code documentation)

##### Description of change

Solves #1181. The original method for file attachment cache validation was to use a max age parameter, which meant that modifying a file but maintaining its filename would cause stale preview images. Using a checksum instead forces the invalidation should the file content change. 

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/viewitem/ItemFilestoreServlet.java [hunk 1]
```diff
@@ -185,17 +185,8 @@ public InitialFilestoreStream(
...
-      // a copy of the resource - the age values only specify how long
-      // before the cache must *revalidate* their copy. With these
-      // settings, browser caches can serve their copy for a whole day
-      // without needing to talk to the server. Shared caches (eg, a Squid
-      // proxy cache) must always revalidate before releasing their copy
-      // to ensure we're not returning a resource to a user who doesn't
-      // have access.
-      return "max-age=86400, s-maxage=0, must-revalidate";
+      // this is to ensure that the etag is checked when attempting to pull down attachments
+      return "must-revalidate";
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/viewitem/ItemFilestoreServlet.java]: Considering ETag is enabled, I wonder if we still need Cache-Control HTTP header.

Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/src/com/tle/web/viewitem/ItemFilestoreServlet.java]: We need the must-revalidate to enforce that the etag check is always made. The max-age and s-maxage is no longer necessary.

**Comment by abidingotter** [comment]
This is probably going to kill the server. It reads the entire file into memory to generate the hash every time. That's why I didn't enable it in general. The real fix would be to generate the hash once upon upload and store it... Somewhere

**Comment by abidingotter** [comment]
I wonder if changing to max-age=0 will make the browser check if the file has changed or not?

**Comment by PenghaiZhang** [comment]
> I wonder if changing to max-age=0 will make the browser check if the file has changed or not?

I may be wrong, but as far as I know, max-age=0 means the cache expires immediately so it enforces the browser to re-validate the resource every time.  Then the server will have to check if the requested resource is latest for every request by either Etag or Last_Modified_date.

**Comment by abidingotter** [comment]
Yep, that's what I mean. The check for last modified date should be super quick.

**Comment by abidingotter** [comment]
We don't calculate etag for anything except the admin console files.

**Pull request merged**